packages/test/test-gc-sweep-tests/src/dataObjectWithCounter.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IRandom } from "@fluid-internal/stochastic-test-utils";
+import { DataObject } from "@fluidframework/aqueduct";
+import { assert } from "@fluidframework/common-utils";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { SharedCounter } from "@fluidframework/counter";
+
+/**
+ * DataObjectWithCounter increments a SharedCounter as a way of sending ops.
+ *
+ * The SharedCounter is retrieved via handle
+ */
+const counterKey = "counter";
+export class DataObjectWithCounter extends DataObject {
+    private _counterHandle?: IFluidHandle<SharedCounter>;
+    public isRunning: boolean = false;
+    public static get type(): string {
+        return "OpSendingDataObject";
+    }
+
+    public get counterHandle(): IFluidHandle<SharedCounter> {
+        assert(this._counterHandle !== undefined, `CounterHandle should always be defined!`);
+        return this._counterHandle;
+    }
+
+    protected async initializingFirstTime(props?: any): Promise<void> {
+        this.root.set<IFluidHandle>(counterKey, SharedCounter.create(this.runtime).handle);
+    }
+
+    protected async hasInitialized(): Promise<void> {
+        this._counterHandle = this.root.get<IFluidHandle<SharedCounter>>(counterKey);
+    }
+
+    public async sendOp(_count: number, _random: IRandom) {
+        assert(this.counterHandle !== undefined, "Can't send ops when counter handle isn't set!");
```

Comment:
this.counterHandle is never undefined as per the getter signature. If it is, the getter will throw an assert and the code will never reach here anyway.